addons/l10n_mx/controllers/portal.py

```diff
@@ -0,0 +1,20 @@
+from odoo.addons.portal.controllers import portal
+from odoo.http import request
+
+class CustomerPortal(portal.CustomerPortal):
+
+    def _get_mandatory_fields(self):
+        try:
+            if request.env['res.country'].sudo().browse(int(request.params['country_id'])).code == 'MX':
+                return super()._get_mandatory_fields() + ['zipcode', 'vat']
+        except ValueError:
+            pass
+        return super()._get_mandatory_fields()
+
+    def _get_optional_fields(self):
+        try:
+            if request.env['res.country'].sudo().browse(int(request.params['country_id'])).code == 'MX':
+                return [field for field in super()._get_optional_fields() if field not in ['zipcode', 'vat']]
+        except ValueError:
+            pass
```

Comment:
only need to try on the conversion of 'country_id' to int (and only if country_id is set), this way we don't catch unexpected errors